File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minprice = prices[0]; // Assuming 1st element as low
        int maxprofit =0;
        for(int i=1; i<prices.size(); i++){
            int cost= prices[i]-minprice;
            maxprofit=max(maxprofit, cost);
            minprice=min(minprice, prices[i]); // Compare the past & current Prices
        }
        return maxprofit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int maxi = 0, mini = prices[0];

        for(int i = 1; i < n; i++){
            int profit = prices[i] - mini;
            maxi = Math.max(maxi, profit);
            mini = Math.min(mini, prices[i]);
        }
        return maxi;
    }
}